src/components/WeatherApp.jsx

```diff
@@ -65,36 +73,61 @@ const WeatherApp = () => {
       }
     );
   };
-
+  const handleMenuSection = () => {
+    menu === "none" ? setMenu("block") : setMenu("none");
+  };
   return (
     <div>
-      <main>
-        <div className="weatherApp">
-          <div className="searchSection">
-            <input
-              type="text"
-              className="inputText"
-              placeholder="Search..."
-              value={inputName}
-              onChange={(e) => setInputName(e.target.value)}
-              onKeyDown={handleInputBoxKeydown}
-            />
-            <div className="location-icon">
-              <FontAwesomeIcon icon={faLocationDot} onClick={handleLocation} />
+      <main className={`main-${theme}`}>
+        <div className={`weatherApp-${theme} weatherApp`}>
+          <div
+            className={` hamburgerMenu-${theme} ${
+              menu === "none" ? "hamburgerMenuHide" : "hamburgerMenuShow"
```

Comment:
this is a good way of making the menu appear or disappear , and it works. 
But a more intuitive way of writing the component render logic is by just anding the component. 
```
const [showSideBar , setShowSidebar]=useState(false)
{showSidebar && <Sidebar />}

```

and when you want to toggle sidebar just toggle the boolean state , dont play with this css. 
logic should reside in JS, styling in CSS